build.gradle

```diff
@@ -33,7 +40,34 @@ tasks.withType(Test) {
     environment 'GRADLE_RELEASE_REPOSITORY','https://services.gradle.org/distributions'
 }
 
+// Should be in buildSrc ideally
+//@CacheableRule
+//abstract class ComposedJarRule implements ComponentMetadataRule {
+//    @Inject abstract ObjectFactory getObjects()
+//
+//    void execute(ComponentMetadataContext context) {
+//        context.details.withVariant("intellijPlatformComposedJar") {
+//            attributes {
+//                attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE, objects.named(LibraryElements, LibraryElements.JAR))
+//            }
+//        }
+//    }
+//}
+
 dependencies {
+    //components {
+    //    withModule("com.redhat.devtools.intellij:intellij-common", ComposedJarRule)
+    //}
+    components {
+        withModule("com.redhat.devtools.intellij:intellij-common") {
+            withVariant("intellijPlatformComposedJar") {
+                attributes {
+                    attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE, objects.named(LibraryElements, LibraryElements.JAR))
+                }
+            }
+        }
+    }
+
     implementation(
             "com.redhat.devtools.intellij:intellij-common:${intellijCommonVersion}",
```

Comment:
I just noticed another api which might work you can try:
```kotlin
implementation("com.redhat.devtools.intellij:intellij-common:${intellijCommonVersion}") {
        attributes {
            attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE, objects.named("composed-jar"))
        }
}
```

Reply:
tested it in a different PR where I needed this: https://github.com/redhat-developer/intellij-kubernetes/pull/781/commits/df398b441f6b4ba5ef6b050452281101bd4b2459 Works super nicely. I'll add you as co-author to my PR. Thanks!